### PR TITLE
New working machine added

### PR DESCRIPTION
**Pull request by algestam**
-----------
Donkey Kong (Nintendo Game & Watch) [algestam]

### CODE REVIEW

File: src/mame/drivers/hh_sm510.cpp [hunk 1]
```diff
@@ -1137,6 +1137,90 @@ MACHINE_CONFIG_END
...
+        MCFG_DEVICE_ADD("maincpu", SM510)
+        MCFG_SM510_R_MASK_OPTION(2) // confirmed
+        MCFG_SM510_WRITE_SEGS_CB(WRITE16(*this, hh_sm510_state, sm510_lcd_segment_w))
+        MCFG_SM510_READ_K_CB(READ8(*this, hh_sm510_state, input_r))
+        MCFG_SM510_WRITE_S_CB(WRITE8(*this, hh_sm510_state, input_w))
+        MCFG_SM510_WRITE_R_CB(WRITE8(*this, hh_sm510_state, piezo_r1_w))
+        MCFG_SM510_READ_B_CB(IOPORT("B"))
+
+        /* video hardware */
+        MCFG_SCREEN_SVG_ADD("screen_top", "svg_top")
```
Comment (happppp) [inline src/mame/drivers/hh_sm510.cpp]: Aspect ratio here is copy-pasted from the dkong2 driver.
If the SVGs are the same size as the ones you sent me earlier today, height for both screens is 1266/2.

File: src/mame/drivers/hh_sm510.cpp [hunk 2]
```diff
@@ -6557,6 +6641,16 @@ ROM_START( exospace )
 	ROM_LOAD( "exospace.svg", 0, 66790, BAD_DUMP CRC(df31043a) SHA1(2d8caf42894df699e469652e5f448beaebbcc1ae) )
 ROM_END
 
+ROM_START( gnw_dkong )
+        ROM_REGION( 0x1000, "maincpu", 0 )
+        ROM_LOAD( "dk-52", 0x0000, 0x1000, CRC(5180cbf8) SHA1(5174570a8d6a601226f51e972bac6735535fe11d) )
+
+        ROM_REGION( 267443, "svg_top", 0)
```
Comment (happppp) [inline src/mame/drivers/hh_sm510.cpp]: Could you make region length same as file length? For the other screen too

File: src/mame/drivers/hh_sm510.cpp (hunk 1 above)
Comment (algestam) [reply src/mame/drivers/hh_sm510.cpp]: Thanks for finding this! I've included this fix in a new commit :)

File: src/mame/drivers/hh_sm510.cpp (hunk 2 above)
Comment (algestam) [reply src/mame/drivers/hh_sm510.cpp]: Same goes for this one, I missed that the filesize was specified two times per file. Thanks for finding it!

**Comment by happppp** [comment]
All good, thanks

**Pull request merged**